core/services/gateway/handlers/functions/allowlist/allowlist.go

```diff
@@ -172,17 +208,113 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 	if err != nil {
 		return errors.Wrap(err, "unexpected error during functions_allow_list.NewTermsOfServiceAllowList")
 	}
-	addrList, err := tosContract.GetAllAllowedSenders(&bind.CallOpts{
+
+	var allowedSenderList []common.Address
+	if !a.config.StoreAllowedSendersEnabled {
```

Comment:
It would be cool if we could still store in the DB when using the old API. That way we could partially launch the feature right away. Could we still update the DB here? Maybe delete everything that's not on the list and then add the whole list? We can discuss offline if you think that's a lot of work.

Reply:
I think its possible, we would need to add a new ORM method to delete everything for a particular `router_contract_address` and store it. I would prefer to add this in a separate PR and close this one. Wdyt?

Reply:
Yes, that's fine, let's do it separately.